client/src/main/java/org/eclipse/hono/client/impl/HonoConnectionImpl.java

```diff
@@ -1043,13 +1043,18 @@ public final void disconnect(final Handler<AsyncResult<Void>> completionHandler)
      * @return The remote container id or {@code null}.
      */
     @Override
-    public String getRemoteContainer() {
+    public String getRemoteContainerId() {
         if (!isConnectedInternal()) {
             return null;
         }
         return connection.getRemoteContainer();
     }
 
+    @Override
+    public String getContainerId() {
+        return connectionFactory.getContainerId();
```

Comment:
FMPOV this should return the (local) container ID of the ProtonConnection instead ...

Reply:
Presuming that we want to use the same container ID for all `ProtonConnection`s created by this `HonoConnectionImpl` (see discussion below):
Using `connection.getContainer()` would complicate things for the scenario here as that would only work if the connection has already been established. I would rather define `HonoConnection.getContainerId()` as the container id that _has been_ or that _will be_ advertised to the remote peer. The implementation would simply return the value of a new `HonoConnectionImpl.containerId` field here.